### PR TITLE
Add support for building on stable Rust on Windows and Unix

### PR DESCRIPTION
**Pull request by fitzgen**
@pepyakin @ZackPierce @DrGoldfire could one of you review this? Thanks :)

### CODE REVIEW

**Comment by ZackPierce** [comment]
Will review tonight or in the morning.

**Comment by ZackPierce** [comment]
The code and documentation changes look reasonable, but the failed CI builds need further investigation. Will have to dig in more tomorrow.

**Comment by ZackPierce** [comment]
Looks like the merged changes did not address the travis failures.

**Comment by fitzgen** [comment]
Rebased and I think that should do the trick. We'll see.

**Comment by fitzgen** [comment]
Thanks @ZackPierce :)

File: .travis.yml [hunk 1]
```diff
@@ -20,3 +20,15 @@ before_script:
...
+
+matrix:
+  include:
+    - rust: stable
+      before_script:
+        - rustup target add i686-pc-windows-gnu
+      script:
+        - cd ./wee_alloc
+        - cargo check --release --features "size_classes"
+        - cargo check --release
```
Comment (RReverser) [inline .travis.yml]: I believe these are wrong - `size_classes` is enabled by default, so one of the lines had to remain `cargo check --release` but another is supposed to be `cargo check --release --no-default-features`.

**Pull request merged**